exporting patch:
# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1506287098 18000
#      Sun Sep 24 16:04:58 2017 -0500
# Node ID fbbbbddc5dd836c35f9dad704062de8eb4328837
# Parent  1265c7f0119a75356f1b4950e9a2f8ec819d81bc
Move worker thread shutdown to main window handling finish (bug #50025)

* main-window.cc (main_window::main_window): Remove octave_finished_signal()
  connection from m_interpreter--which runs in m_main_thread--to
  m_main_thread's quit().
  (main_window::handle_octave_finished): Quit and wait on the worker thread.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- libgui/src/main-window.cc
+++ libgui/src/main-window.cc
@@ -217,9 +217,6 @@ main_window::main_window (QWidget *p, oc
   connect (m_interpreter, SIGNAL (octave_finished_signal (int)),
            this, SLOT (handle_octave_finished (int)));
 
-  connect (m_interpreter, SIGNAL (octave_finished_signal (int)),
-           m_main_thread, SLOT (quit (void)));
-
   connect (m_main_thread, SIGNAL (finished (void)),
            m_main_thread, SLOT (deleteLater (void)));
 
@@ -1579,6 +1576,9 @@ main_window::handle_octave_ready (void)
 void
 main_window::handle_octave_finished (int exit_status)
 {
+  m_main_thread->quit();
+  m_main_thread->wait();
+
   qApp->exit (exit_status);
 }
 
